instrumentation/redisson-3.0/src/test/groovy/RedissonClientTest.groovy

```diff
@@ -0,0 +1,303 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.opentelemetry.auto.test.AgentTestRunner
+import io.opentelemetry.auto.test.utils.PortUtils
+import io.opentelemetry.trace.attributes.SemanticAttributes
+import org.redisson.Redisson
+import org.redisson.api.RAtomicLong
+import org.redisson.api.RBatch
+import org.redisson.api.RBucket
+import org.redisson.api.RList
+import org.redisson.api.RLock
+import org.redisson.api.RMap
+import org.redisson.api.RScoredSortedSet
+import org.redisson.api.RSet
+import org.redisson.api.RedissonClient
+import org.redisson.config.Config
+import redis.embedded.RedisServer
+import spock.lang.Shared
+
+import static io.opentelemetry.trace.Span.Kind.CLIENT
+
+class RedissonClientTest extends AgentTestRunner {
+
+  @Shared
+  int port = PortUtils.randomOpenPort()
+
+  @Shared
+  RedisServer redisServer = RedisServer.builder()
+  // bind to localhost to avoid firewall popup
+    .setting("bind 127.0.0.1")
+  // set max memory to avoid problems in CI
+    .setting("maxmemory 128M")
+    .port(port).build()
+  @Shared
+  RedissonClient redisson
+
+  def setupSpec() {
+    println "Using redis: $redisServer.args"
+    redisServer.start()
+  }
+
+  def cleanupSpec() {
+    redisson.shutdown()
+    redisServer.stop()
+  }
+
+  def setup() {
+    Config config = new Config()
+    config.useSingleServer().setAddress("localhost:" + port)
+    redisson = Redisson.create(config)
+    TEST_WRITER.clear()
+  }
+
+  def "test string command"() {
+    when:
+    RBucket<String> keyObject = redisson.getBucket("foo")
+    keyObject.set("bar")
+    keyObject.get()
+    then:
+    assertTraces(2) {
+      trace(0, 1) {
+        span(0) {
+          operationName "SET"
+          spanKind CLIENT
+          attributes {
+            "${SemanticAttributes.DB_SYSTEM.key()}" "redis"
+            "${SemanticAttributes.NET_PEER_IP.key()}" "127.0.0.1"
+            "${SemanticAttributes.NET_PEER_NAME.key()}" "localhost"
+            "${SemanticAttributes.DB_CONNECTION_STRING.key()}" "localhost:$port"
+            "${SemanticAttributes.NET_PEER_PORT.key()}" port
+            "${SemanticAttributes.DB_STATEMENT.key()}" "SET"
+          }
+        }
+      }
+      trace(1, 1) {
+        span(0) {
+          operationName "GET"
+          spanKind CLIENT
+          attributes {
+            "${SemanticAttributes.DB_SYSTEM.key()}" "redis"
+            "${SemanticAttributes.NET_PEER_IP.key()}" "127.0.0.1"
+            "${SemanticAttributes.NET_PEER_NAME.key()}" "localhost"
+            "${SemanticAttributes.DB_CONNECTION_STRING.key()}" "localhost:$port"
+            "${SemanticAttributes.NET_PEER_PORT.key()}" port
+            "${SemanticAttributes.DB_STATEMENT.key()}" "GET"
+          }
+        }
+      }
+    }
+  }
+
+  def "test batch command"() {
+    when:
```

Comment:
Does it make sense to have a more complete async test with callbacks? I notice the return type is `RFuture` I wonder if our concurrent instrumentation will apply to it.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/lettuce/lettuce-5.1/src/test/groovy/io/opentelemetry/instrumentation/auto/lettuce/v5_1/LettuceAsyncClientTest.groovy#L327

We can focus on async in another PR though.

Reply:
Nothing seems to have changed.